HW3/P5/label_regions.cl

```diff
@@ -78,22 +77,58 @@ propagate_labels(__global __read_write int *labels,
     // Fetch the value from the buffer the corresponds to
     // the pixel for this thread
     old_label = buffer[buf_y * buf_w + buf_x];
-
+    
     // CODE FOR PARTS 2 and 4 HERE (part 4 will replace part 2)
+    // Only the first kernel in the workgroup should execute the 
+    // grandparent loading code
+    if (idx_1D == 0) {
+        int last_label_id = 0;
+        int last_label = 0;
+        for (int row = 0; row < get_local_size(0); row++) {
+            for (int col = 0; col < get_local_size(1); col++) {
+                int t_buf_x = col + halo;
+                int t_buf_y = row + halo;
+                int t_pos = t_buf_y * buf_w + t_buf_x;
+                int t_old_label = buffer[t_pos];
+                // Check to see if this is the same 
+                // index of the position we just loaded
+                if (t_old_label != last_label_id) {
```

Comment:
Need to make sure that t_old_label isn't w*h, or otherwise you'll be accessing a wrong value.